src/components/video-controller.js

```diff
@@ -66,91 +70,102 @@ export class VideoController {
         this.stopControlBarTimer();
     }
 
-    showVideo(visible) {
-        this.video.style.visibility = visible ? 'visible' : 'hidden';
-
-        // If we are showing the video, we are loaded enough.
-        if (visible) this.showLoadingSpinner(false);
-    }
-
     showLoadingSpinner(visible) {
         const spinner = this.loadingSpinner;
         if (!spinner) return;
         if (visible) spinner.show();
         else spinner.hide();
     }
 
-    startVideo(videoStream) {
-        this.videoStarted = true;
+    // Create the video element "later" to work around some hangs and crashes, e.g. on the PS4
+    startVideoLater(videoStream, showControlBar) {
+        this.stopOldVideo(videoStream);
+        setTimeout(() => this.startVideo(videoStream, showControlBar), 1);
+    }
+
+    startVideo(videoStream, showControlBar) {
+        this.stopOldVideo(videoStream);
+
+        const isFirstStart = !!videoStream;
+        if (videoStream) {
+            this.videoStream = videoStream;
+        } else {
+            videoStream = this.videoStream;
+        }
+        if (!videoStream) return;
```

Comment:
I wonder why we are not using this.videoStream instead of videoStream?
I am sure it has something to do with the GC?